bundles/org.eclipse.equinox.simpleconfigurator/src/org/eclipse/equinox/internal/simpleconfigurator/ConfigApplier.java

```diff
@@ -463,21 +439,12 @@ private void refreshPackages(Bundle[] bundles, BundleContext context) {
 				latch.countDown();
 			}
 		};
-		context.addFrameworkListener(listener);
-		packageAdminService.refreshPackages(allSameBSNs.toArray(new Bundle[0]));
-
+		frameworkWiring.refreshBundles(bundles, listener);
 		try {
 			latch.await();
```

Comment:
How is this not blocking forever if you never add the listener.  This makes me believe there are no tests for any of these changes you make here.

Reply:
`frameworkWiring.refreshBundles( ... )` accepts the listener as an argument here, I just copied it from the existing code you added here

 https://github.com/eclipse-equinox/equinox/commit/907ec00a829e2fe03f156089775c70ddac834d6e

or do I misunderstood you comment?

Reply:
OK, got it.  I missed the listener being passed in.